src/main/java/de/adorsys/keycloak/config/repository/UserProfileRepository.java

```diff
@@ -0,0 +1,94 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package de.adorsys.keycloak.config.repository;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.adorsys.keycloak.config.exception.KeycloakRepositoryException;
+import org.apache.commons.lang3.StringUtils;
+import org.keycloak.admin.client.resource.UserProfileResource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Optional;
+import javax.ws.rs.core.Response;
+
+@Component
+public class UserProfileRepository {
+
+    private final RealmRepository realmRepository;
+
+    private final ObjectMapper mapper;
+
+    @Autowired
+    public UserProfileRepository(RealmRepository realmRepository, ObjectMapper mapper) {
+        this.realmRepository = realmRepository;
+        this.mapper = mapper;
+    }
+
+    public void updateUserProfile(String realm, String newPprofileResourceConfiguration) throws JsonProcessingException {
+
+        var realmAttributes = realmRepository.get(realm).getAttributesOrEmpty();
+        var userProfileEnabled = Boolean.valueOf(realmAttributes.getOrDefault("userProfileEnabled", "false"));
+        var userProfileResource = getResource(realm);
+
+        if (!userProfileEnabled) {
+            userProfileResource.update(null);
+            return;
+        }
+
+        if (userProfileEnabled && userProfileResource == null) {
+            return;
+        }
+
+        String profileResourceConfiguration = Optional.ofNullable(userProfileResource.getConfiguration()).orElse("");
+        var profileDefintionHasNotChanged = hasProfileDefinitionChanged(newPprofileResourceConfiguration, profileResourceConfiguration);
+        if (profileDefintionHasNotChanged) {
+            return;
+        }
+
+        var updateUserProfileResponse = userProfileResource.update(newPprofileResourceConfiguration);
+        if (!updateUserProfileResponse.getStatusInfo().equals(Response.Status.OK)) {
+            throw new KeycloakRepositoryException("Could not update UserProfile Definition");
+        }
+
+    }
+
+    private boolean hasProfileDefinitionChanged(String newPprofileResourceConfiguration, String profileResourceConfiguration)
+            throws JsonProcessingException {
+        if (StringUtils.isEmpty(newPprofileResourceConfiguration)) {
+            newPprofileResourceConfiguration = "";
+        }
+        if (StringUtils.isEmpty(profileResourceConfiguration)) {
+            profileResourceConfiguration = "";
+        }
+
+        var currentValue = mapper.readValue(profileResourceConfiguration, Map.class);
+        var newValue = mapper.readValue(newPprofileResourceConfiguration, Map.class);
```

Comment:
I just gave this a try and noticed that the `ObjectMapper` is not provided as a bean if we run the cli tool.
It seems that other places of the code use the Methods from `de.adorsys.keycloak.config.util.JsonUtil` to use a shared object mapper.

Suggestion use:
```
var currentValue = JsonUtil.fromJson(profileResourceConfiguration, Map.class);
var newValue = JsonUtil.fromJson(newPprofileResourceConfiguration, Map.class);
```

and add a method like:
```
    public static <T> T fromJson(String value, Class<T> type) {
        try {
            return objectMapper.readValue(value, type);
        } catch (JsonProcessingException e) {
            throw new ImportProcessingException(e);
        }
    }
```

After that you can remove the objectMapper constructor parameter and field (and unused imports).